From 859698d29b547217356851094ed8188236e717b6 Mon Sep 17 00:00:00 2001
From: Volker Lendecke <vl@samba.org>
Date: Sat, 3 Feb 2018 07:07:24 +0100
Subject: [PATCH] libads: Fix the build --without-ads

Signed-off-by: Volker Lendecke <vl@samba.org>
Reviewed-by: David Disseldorp <ddiss@samba.org>

Autobuild-User(master): David Disseldorp <ddiss@samba.org>
Autobuild-Date(master): Tue Feb  6 02:47:44 CET 2018 on sn-devel-144
---
 source3/libads/kerberos_keytab.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/source3/libads/kerberos_keytab.c b/source3/libads/kerberos_keytab.c
index ffd100c5636..37ac7ba985e 100644
--- a/source3/libads/kerberos_keytab.c
+++ b/source3/libads/kerberos_keytab.c
@@ -640,7 +640,11 @@ int ads_keytab_list(const char *keytab_name)
 	}
 
 	if (keytab_name == NULL) {
+#ifdef HAVE_ADS
 		ret = ads_keytab_open(context, &keytab);
+#else
+		ret = ENOENT;
+#endif
 	} else {
 		ret = smb_krb5_kt_open(context, keytab_name, False, &keytab);
 	}
-- 
2.14.1

